src/constructor.io/models/common/IFilterable.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+
+namespace Constructorio_NET.Models
+{
+    public interface IFilterable
+    {
+        /// <summary>
+        /// Gets or sets filters used to refine results.
+        /// </summary>
+        Dictionary<string, List<string>> Filters { get; set; }
+
+        /// <summary>
+        /// Gets or sets user test cells.
+        /// </summary>
+        Dictionary<string, string> TestCells { get; set; }
+
+        /// <summary>
+        /// Gets or sets collection of user related data.
+        /// </summary>
+        UserInfo UserInfo { get; set; }
+
+        /// <summary>
+        /// Gets or sets how to return variation data.
+        /// </summary>
+        VariationsMap VariationsMap { get; set; }
```

Comment:
Should this be under the `IFilterable` interface? When I think of a class having "IFilterable", I wouldn't imagine it having/needing to define things like `UserInfo` or `TestCells`. Maybe a separate interface like `IUserParameters` would make more sense?

`VariationsMap` might make more sense to be together with `FmtOptions` but I can see it included here.

`preFilterExpressions` seems like something that belongs here, esp. if we're including it in Autocomplete but then we'd probably have to include an implementation for it since it's currently missing. We might need to do it though if we're releasing an interface.